### ISSUE TITLE
Saved media not paginating

### ISSUE THREAD

**Issue by khaliddd** [sbertix/Swiftagram#169]
Hi @sbertix, 

I am getting parsing error in saved posts endpoint, My code is: 

```
Endpoint.Media.Posts.saved(startingAt: nextId)
            .unlocking(with: secret)
            .task(maxLength: 4) { (page) in
                completion(posts, next, error)
            } onChange: { (response) in
                if case .success(let collection) = response, let sposts = collection.media {
                    posts.append(contentsOf: sposts)
                    next = collection.pagination.next
                }
                else if case .failure(let e) = response {
                    error = e
                }
            }.resume()
```

it looks like it fetches the correct data but can't not parse it. 

while the end point to get Liked posts is working fine.

**Comment by khaliddd** [comment]
Hey @sbertix , 

there is a bug in Liked endpoint as well 
`Endpoint.Media.Posts.liked`

pagination parameter is nil every time , even if there are more pages.

**Comment by sbertix** [comment]
Hey @khaliddd, 

I'm gonna inspect it and fix it for `5.0.0`. 
Thx 💪

**Comment by sbertix** [comment]
I'm not sure why this happens cause Instagram did not change the pagination parameters. 
It might be related to an issue with the `4.*` pagination offsets, but for sure it works with `5.*` (which I'm planning on releasing by the end of the week, btw). 

I'm definitely adding `saved`-specific models, though, as that has indeed changed since I've initially implemented it. Meaning with `5.*` you'd be able to access "saved albums", etc.

**Comment by khaliddd** [comment]
Hey @sbertix : 

I found another thing, pagination in `Endpoint.Friendship.followed` is also not working (while its working fine in `Endpoint.Friendship.following`)

can you please check it as well.

**Comment by sbertix** [comment]
Do you mean the pagination parameter offset is `nil` on the model, or it doesn't actually work? 
Cause if it's the model, don't worry, it's been fixed. If it's about the pagination, as it hasn't changed, I'm assuming it might be related to the account you're using for tests being rate limited.

**Comment by khaliddd** [comment]
`collection.pagination` is nil. I guess its not related to limit, as its working fine in `Endpoint.Friendship.following`

**Comment by sbertix** [comment]
I mean, not necessarily. It's not unlikely for them to stop working asymmetrically. 
What I'm asking though is: if you let the `followed` paginate more than one will it load all requested pages? Is it just an issue with the `pagination` parameter or the actual response?

**Comment by khaliddd** [comment]
yes it fetches all requested pages, I tried with 2,3,4 and 5. 
but `pagination` is nil in all cases

**Comment by sbertix** [comment]
Then it's okay, it just the model offset. 
You can still access it from the underlying `Wrapper`, and it won't be a problem in `5.0.0`.

**Comment by khaliddd** [comment]
okay, thank you

**Comment by khaliddd** [comment]
@sbertix : in `Endpoint.Friendship.followed` ,  I tried to use nextMaxId from wrapper, but I am not getting correct value from wrapper as well,

Thats what from wrapper : `"nextMaxId": 50`

do you know when you'll release next version?

**Comment by sbertix** [comment]
The next max id issue is not related to actual changes in Insta endpoints, it's about new limitations with usage. Nothing I can do, unfortunately. 

I hope to release this week. 
It's a huge update and it's taking longer than I hoped for to polish it.

**Comment by khaliddd** [comment]
Thank you for the quick response, Yeah I understand instagram limitation, but I guess you did not get my point, let me explain: 
if I call the api like this : 

```
Endpoint.Friendship.followed(by: userId)
            .unlocking(with: secret)
            .task(maxLength: 15) { (page) in
               
            } onChange: { (response) in
                
            }.resume()
```

it works fine and it returns 15 pages (700+ users), but if I call the api for 2 pages, then it returns 2 pages but pagination is nil, but there should be a valid pagination value, as there are 13 more pages.

**Comment by sbertix** [comment]
> Thank you for the quick response, Yeah I understand instagram limitation, but I guess you did not get my point, let me explain:
> if I call the api like this :
> 
> ```
> Endpoint.Friendship.followed(by: userId)
>             .unlocking(with: secret)
>             .task(maxLength: 15) { (page) in
>                
>             } onChange: { (response) in
>                 
>             }.resume()
> ```
> 
> it works fine and it returns 15 pages (700+ users), but if I call the api for 2 pages, then it returns 2 pages but pagination is nil, but there should be a valid pagination value, as there are 13 more pages.

I was replying to you getting a `nextMaxId` of `50`, and that resulting in wrong pagination. 
What you're talking about now is something I've already told you is fixed in `5.0.0` and has nothing to do with your last comment 😊
